servlet/src/main/java/io/undertow/servlet/spec/ServletContextImpl.java

```diff
@@ -207,11 +207,12 @@ public Void call(HttpServerExchange exchange, ThreadSetupHandler.Action<Void, Ob
 
     @Override
     public String getContextPath() {
-        String contextPath = deploymentInfo.getContextPath();
-        if (contextPath.equals("/")) {
-            return "";
-        }
-        return contextPath;
+            checkDeploymentInfo();
```

Comment:
This type of approach does not work to prevent the NPE. The reason is that we risk having the deploymentInfo set to ull between the execution of checkDeploymentInfo and the next line.